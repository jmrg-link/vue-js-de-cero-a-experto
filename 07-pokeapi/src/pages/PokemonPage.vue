<template lang="">
    <div>
        <h1>Â¿Quien es este pokemon?</h1>
        <!--TODO: img --><PokemonPicture />
        <!--TODO: opciones --><PokemonOptions />
    </div>
</template>
<script>
import PokemonOptions from '@/components/PokemonOptions'
import PokemonPicture from '@/components/PokemonPicture'

export default {
    components: {
        PokemonPicture,
        PokemonOptions
    }
}
</script>
