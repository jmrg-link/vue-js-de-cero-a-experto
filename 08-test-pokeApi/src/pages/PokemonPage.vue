<!-- eslint-disable -->
<template>
  <h1 v-if="!pokemon">Espere por favor...</h1>
  <div v-else>
    <h1>Â¿Quien es este pokemon?</h1>
    <PokemonPicture :pokemonId="pokemon.id" :showPokemon="showPokemon" />
    <PokemonOptions :pokemons="pokemonArr" @selection="checkAnswer($event)" />
    <template v-if="showAnswer">
      <h2 class="fade-in">{{ message }}</h2>
      <button @click="newGame">New Game</button>
    </template>
  </div>
</template>
<script>
import PokemonOptions from "@/components/PokemonOptions";
import PokemonPicture from "@/components/PokemonPicture";
import getPokemonOptions from "../helpers/getPokemonOptions";

//console.log(getPokemonOptions())

export default {
  components: {
    PokemonOptions,
    PokemonPicture,
  },
  data() {
    return {
      pokemonArr: [],
      pokemon: "",
      showPokemon: false,
      showAnswer: false,
      message: "",
    };
  },
  methods: {
    async mixPokemonArray() {
      this.pokemonArr = await getPokemonOptions();
      const rndInt = Math.floor(Math.random() * 4);
      this.pokemon = this.pokemonArr[rndInt];
      console.log(this.pokemon);
    },
    checkAnswer(selectedId) {
      this.showPokemon = true;
      this.showAnswer = true;
      if (selectedId === this.pokemon.id) {
        this.message = `Correcto , ${this.pokemon.name}`;
      } else {
        this.message = `Nooop.. el pokemon era ${this.pokemon.name} `;
      }
      //console.log(pokemonId);
    },
    newGame(){
      this.showPokemon = false
      this.showAnswer = false
      this.pokemonArr = []
      this.pokemon= null
      this.mixPokemonArray()

    }
  },
  mounted() {
    this.mixPokemonArray();
  },
};
</script>
