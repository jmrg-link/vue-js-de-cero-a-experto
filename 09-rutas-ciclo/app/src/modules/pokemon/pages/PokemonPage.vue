<template>
    <h1>Pokemon Page</h1>
</template>