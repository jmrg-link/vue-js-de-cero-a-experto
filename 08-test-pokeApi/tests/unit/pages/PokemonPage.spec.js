import { shallowMount, mount } from "@vue/test-utils";
import {pokemons} from '../mocks/pokemons.mock'
import PokemonPage from "@/pages/PokemonPage";

describe("PokemonPage component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(PokemonPage);
  });

  test("debe de hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  test("debe de llamar al mixPokemonArray al montar ", () => {
    const mixPokemonArray = jest.spyOn(PokemonPage.methods, "mixPokemonArray");
    shallowMount(PokemonPage);
    expect(mixPokemonArray).toHaveBeenCalled();
  });

  test("debe hacer match cuando cargan los pokemons", () => {
    const wrapper = mount(PokemonPage, {
      data() {
        return {
          pokemonArr: pokemons,
          pokemon: pokemons[0],
          showPokemon: false,
          showAnswer: false,
          message: "",
        }
      }
    })
    console.log(pokemons[0]);
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('debe de mostrar los componentes de pokemonpicture y pokemonoptions ', () => {
    const wrapper = shallowMount(PokemonPage, {
        data() {
          return {
            pokemonArr: pokemons,
            pokemon: pokemons[0],
            showPokemon: false,
            showAnswer: false,
            message: "",
          }
        }
      })

      const picture = wrapper.find('pokemon-picture-stub')
      const options = wrapper.find('pokemon-options-stub')

      expect(picture.exists() ).toBeTruthy()
      expect(options.exists() ).toBeTruthy()

      expect(picture.attributes('pokemonid')).toBe('5')
      expect(options.attributes('pokemons')).toBeTruthy()



      //PokemonOptions attribute pokemons toBe true
  })

  test('test con checkAnswer ', async() => {
    const wrapper = shallowMount(PokemonPage, {
        data() {
          return {
            pokemonArr: pokemons,
            pokemon: pokemons[0],
            showPokemon: false,
            showAnswer: false,
            message: "",
          }
        }
      })

     // console.log(wrapper.vm)
    await wrapper.vm.checkAnswer(5)
    //console.log(wrapper.find('h2').text()) 
    expect(wrapper.find('h2').exists()).toBeTruthy()
    expect(wrapper.vm.showPokemon).toBe(true)
    expect(wrapper.find('h2').text()).toBe(`Correcto , ${pokemons[0].name}`)

    await wrapper.vm.checkAnswer(10)
    expect(wrapper.vm.message).toBe(`Nooop.. el pokemon era ${pokemons[0].name}`)
  })

})
