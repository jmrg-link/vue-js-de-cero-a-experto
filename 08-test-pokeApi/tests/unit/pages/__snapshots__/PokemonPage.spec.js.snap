// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPage component debe de hacer match con el snapshot 1`] = `<h1>Espere por favor...</h1>`;

exports[`PokemonPage component debe hacer match cuando cargan los pokemons 1`] = `
<div>
  <h1>Â¿Quien es este pokemon?</h1>
  <div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/5.svg" class="hidden-pokemon" alt="pokemon"></div>
  <div class="options-container">
    <ul>
      <li>Pikachu</li>
      <li>charmander</li>
      <li>venusaur</li>
      <li>mew</li>
    </ul>
  </div>
  <!--v-if-->
</div>
`;
